force-app/main/default/classes/ServiceDeliveryService.cls

```diff
@@ -232,18 +237,21 @@ public with sharing class ServiceDeliveryService {
         return (ServiceDelivery__c) deliveryRecord;
     }
 
-    private Map<String, List<String>> getStatusBuckets() {
-        List<String> bucketNames = new List<String>{ ABSENT, PRESENT };
+    public Map<String, List<String>> getStatusBuckets(
```

Comment:
I'm not sure the updates to this class are necessary - I think they could reverted without altering the rest of the delivery here. I also prefer the restricted argument list to "getStatusBuckets" because I think that the ServiceDeliveryService should really be responsible for delivering bucket info for the ServiceDelivery sobject only, and the method all already narrows the scope to the status field.